### PR TITLE
C_enums

### LINKED ISSUE #1872: [Concept Entry] C Enums
### Reason/inspiration (optional)

We would like a new term entry for [C](https://www.codecademy.com/resources/docs/c) about Enums.

The entry should go in a new file in docs/content/c/concepts/enums/enums.md

The entry should:

- Describe what an enum is.
- Have a syntax section illustrating the creation of an enum with a  ` ```pseudo ` code block.
- An example showing an enum in use in a ` ```c ` code block.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by RodMe27**
### Description

This document is the first documentation of the keyword "enum" for the programming lenguage C.
Closes #1872

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/c/concepts/enums/enums.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
+# Enum
```
Comment (SSwiniarski) [inline content/c/concepts/enums/enums.md:1]: This entry is missing the required metadata. To understand what to put here, please read the [metadata section in the style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md#metadata). The metadata should look like this:

```
---
Title: 'The Title'
Description: 'A brief description'
Subjects:
  - 'A subject name'
  - 'A second subject name'
  - 'An nth subject name'
Tags:
  - 'A tag'
  - 'A second tag'
  - 'An nth tag'
CatalogContent:
  - 'learn-example-course'
  - 'path/example-path'
---

Also, we do not have h1 headers to title Doc entries, the title in the metadata does this automatically.

File: content/c/concepts/enums/enums.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	Wednesday,
+	Thursday,
+	Friday,
+	Saturday
+};
+```
+
+`Sunday` will be setvalue 0, `Monday` will be set to a value of 1, and so  `Patient` structure.
+
+## Another Example
```
Comment (SSwiniarski) [inline content/c/concepts/enums/enums.md:38]: You don't need these two examples. The second example below illustrates the same thing. I'd remove these first two, and move that second example below so it's the first example.

File: content/c/concepts/enums/enums.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+}
+```
+
+### Output
+
+```c
+0 1 2 3 4 5 6 7 8 9 10 11
+```
+
+## Note
```
Comment (SSwiniarski) [inline content/c/concepts/enums/enums.md:65]:
```suggestion
## Examples 
```

File: content/c/concepts/enums/enums.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+### 1. Two enum names can have the same value. For instance, both ‘Running’ and ‘In_Progress’ have the same value of 1.
```
Comment (SSwiniarski) [inline content/c/concepts/enums/enums.md:67]:
```suggestion
Two `enum` names can have the same value. For instance, both `Running` and `In_Progress` have the same value of 1.
```
These are not appropriate as headers.

File: content/c/concepts/enums/enums.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+#include <stdio.h>
+enum State {Todo = 0, Running = 1, In_Progress = 1};
+int main()
+{
+    printf("%d, %d, %d", Todo, Running, In_Progress);
+    return 0;
+}
+```
+
+#### Output
```
Comment (SSwiniarski) [inline content/c/concepts/enums/enums.md:79]:
```suggestion
This example results in the following output.
```

File: content/c/concepts/enums/enums.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+```shell
+0, 1, 1
+```
+
+### 2. If we do not explicitly assign values to enum MemberName, the compiler by default assigns values starting from 0. For instance, Sunday gets a value of 0, Monday gets 1, and so on.
```
Comment (SSwiniarski) [inline content/c/concepts/enums/enums.md:85]:
```suggestion
If we do not explicitly assign values to `enum` `day`, the compiler by default assigns values starting from 0. For instance, `sunday` gets a value of 0, `monday` gets 1, and so on.
```

Can we also move this so it is the first example?

File: content/c/concepts/enums/enums.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+ 
+int main()
+{
+    enum day d = wednesday;
+    printf("The day number stored in d is %d", d);
+    return 0;
+}
+```
+
+#### Output
```
Comment (SSwiniarski) [inline content/c/concepts/enums/enums.md:99]:
```suggestion
This example results in the following output:
```

File: content/c/concepts/enums/enums.md [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+```shell
+The day number stored in d is 3
+```
+
+### 3. We can assign values to a particular name in any order. All unassigned names get value as the value of the previous name plus one.
```
Comment (SSwiniarski) [inline content/c/concepts/enums/enums.md:105]:
```suggestion
We can assign values to a particular name in any order. All unassigned names get value as the value of the previous name plus one.
```

File: content/c/concepts/enums/enums.md [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+ 
+int main()
+{
+    printf("%d %d %d %d %d %d %d", sunday, monday, tuesday,
+            wednesday, thursday, friday, saturday);
+    return 0;
+}
+```
+
+#### Output
```
Comment (SSwiniarski) [inline content/c/concepts/enums/enums.md:120]:
```suggestion
This example results in the following output:
```

File: content/c/concepts/enums/enums.md [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+
+```shell
+1 2 20 21 10 11 12
+```
+
+### 4.  All enum constants must be unique in their scope.
```
Comment (SSwiniarski) [inline content/c/concepts/enums/enums.md:126]:
```suggestion
> **Note:**  All `enum` constants must be unique in their scope.
```

**Review by SSwiniarski — COMMENTED** [review]
@RodMe27, I did a first review and left a few comments.

**Comment by RodMe27** [comment]
@SSwiniarski changes has been updated and push to the branch.

File: content/c/concepts/enums/enums.md [hunk 11]
```diff
@@ -0,0 +1,93 @@
+---
+Title: 'Enum'
+Description: 'Enum is used to assign custom names to integral constants'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Enums'
+  - 'Variable Type'
```
Comment (SSwiniarski) [inline content/c/concepts/enums/enums.md:9]:
```suggestion
  - 'Variable Types'
```

File: content/c/concepts/enums/enums.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
+---
+Title: 'Enum'
+Description: 'Enum is used to assign custom names to integral constants'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Enums'
```
Comment (SSwiniarski) [inline content/c/concepts/enums/enums.md:8]:
```suggestion
  - 'Enum'
```

File: content/c/concepts/enums/enums.md [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+---
+
+A **enum** (enumeration) is a user-defined data type in C. It is used to assign names to integral constants, the principal purpose of this names is to create a program that is easy to read and maintain. It is defined using the `enum` keyword.
+
+## Syntax
+
+```pseudo
+enum name
+{
+	memberName1 = <integral constant>,
```
Comment (SSwiniarski) [inline content/c/concepts/enums/enums.md:22]:
```suggestion
  memberName1 = <integral constant>,
```

File: content/c/concepts/enums/enums.md [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+	memberName2,
```
Comment (SSwiniarski) [inline content/c/concepts/enums/enums.md:23]:
```suggestion
  memberName2,
```

File: content/c/concepts/enums/enums.md [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+	memberName3,
```
Comment (SSwiniarski) [inline content/c/concepts/enums/enums.md:24]:
```suggestion
  memberName3,
```

File: content/c/concepts/enums/enums.md [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+	...
```
Comment (SSwiniarski) [inline content/c/concepts/enums/enums.md:25]:
```suggestion
  ...
```

File: content/c/concepts/enums/enums.md [hunk 17]
```diff
@@ -0,0 +1,93 @@
...
+	memberNameN
```
Comment (SSwiniarski) [inline content/c/concepts/enums/enums.md:26]:
```suggestion
  memberNameN
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@RodMe27 a few more comments.

**Comment by RodMe27** [comment]
@SSwiniarski New changes review committed.

File: content/c/concepts/enums/enums.md [hunk 18]
```diff
@@ -0,0 +1,93 @@
+---
+Title: 'Enum'
+Description: 'Enum is used to assign custom names to integral constants'
```
Comment (SSwiniarski) [inline content/c/concepts/enums/enums.md:3]:
```suggestion
Description: 'An enum is used to assign custom names to integral constants.'
```

File: content/c/concepts/enums/enums.md [hunk 19]
```diff
@@ -0,0 +1,93 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Enum'
+  - 'Variable Types'
+CatalogContent:
+  - 'learn-c'
+  - 'paths/computer-science'
+---
+
+A **enum** (enumeration) is a user-defined data type in C. It is used to assign names to integral constants, the principal purpose of this names is to create a program that is easy to read and maintain. It is defined using the `enum` keyword.
```
Comment (SSwiniarski) [inline content/c/concepts/enums/enums.md:15]:
```suggestion
An **enum** (enumeration) is a user-defined data type in C. It is used to assign names to integral constants. The principal purpose of these names is to create a program that is easy to read and maintain. It is defined using the `enum` keyword.
```

File: content/c/concepts/enums/enums.md [hunk 20]
```diff
@@ -0,0 +1,93 @@
...
+  ...
+  memberNameN
+};
+```
+
+An enum is made up of members where each `memberName` can have a defined integral constant (i.e. 0, 10, 2) or just the `memberName`.
+
+## Examples
+
+If we do not explicitly assign values to `enum` `day`, the compiler by default assigns values starting from 0. For instance, `sundays` gets a value of 0, `monday` gets 1, and so on.
```
Comment (SSwiniarski) [inline content/c/concepts/enums/enums.md:34]:
```suggestion
If we do not explicitly assign values to `enum` `day`, the compiler by default assigns values starting from 0. For instance, `sunday` gets a value of 0, `monday` gets 1, and so on.
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@RodMe27, last few comments.

**Comment by RodMe27** [comment]
Changes committed, thanks for the comments @SSwiniarski

**Review by SSwiniarski — APPROVED** [review]
@RodMe27 Looks good for a second review.

File: content/c/concepts/enums/enums.md [hunk 21]
```diff
@@ -0,0 +1,93 @@
...
+{
+  memberName1 = <integral constant>,
+  memberName2,
+  memberName3,
+  ...
+  memberNameN
+};
+```
+
+An enum is made up of members where each `memberName` can have a defined integral constant (i.e. 0, 10, 2) or just the `memberName`.
```
Comment (caupolicandiaz) [inline content/c/concepts/enums/enums.md:30]:
```suggestion
An enum is made up of members where each `memberName` can have a defined integral constant (e.g. 0, 10, 2) or just the `memberName`.
```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@RodMe27 I've completed my initial review. Please update the entry, and tag me when you've finished making changes. Thanks

**Comment by caupolicandiaz** [comment]
@RodMe27 It's been over a week since I posted the review on this PR, do you need any assistance with the changes?

**Comment by SSwiniarski** [comment]
@RodMe27 in the interest of getting this merged, I addressed @caupolicandiaz's comments myself. @caupolicandiaz, do we look good to merge this now?

**Review by caupolicandiaz — APPROVED** [review]
@RodMe27 @SSwiniarski Looks good to merge. I just made a couple minor edits.

**Comment by SSwiniarski** [comment]
@RodMe27 Congrats on the new entry. Live link here: https://www.codecademy.com/resources/docs/c/enums

image

**Pull request merged**